pkg/promutil/prometheus.go

```diff
@@ -194,6 +195,12 @@ func PromStringTag(text string, labelsSnakeCase bool) (bool, string) {
 
 // sanitize replaces some invalid chars with an underscore
 func sanitize(text string) string {
+
+	// metrics starting with a digit violate the prometheus metric naming convention, so we add an underscore
+	if matched, _ := regexp.MatchString(`^\d`, text); matched {
```

Comment:
Please create a package level var with [`regexp.MustCompile()`](https://pkg.go.dev/regexp#MustCompile) for performance reasons. With this, we take the `regexp.Compile()` performance hit every call.

```go
var metricNumberPrefixRE = regexp.MustCompile(`^\d`)
```

```suggestion
	if matched, _ := metricNumberPrefixRE.MatchString(`^\d`, text); matched {
```

Reply:
Yup. Addressed